src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,126 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final int GROW_FACTOR = 2;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Object[] entries;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        entries = new Object[INITIAL_CAPACITY];
+        threshold = (int)(INITIAL_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getIndex(key);
+        if (entries[index] == null) {
+            addNewEntry(key, value, index);
+        } else {
+            addOrUpdateEntry(key, value, index);
+        }
+        if (size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        if (entries[index] == null) {
+            return null;
+        }
+        return getValue(key, index);
+    }
+
+    @SuppressWarnings("unchecked")
+    private V getValue(K key, int index) {
+        Entry<K, V> entry = (Entry<K, V>) entries[index];
+        while (true) {
+            if (checkKeyEquality(key, entry.key)) {
+                return entry.value;
+            }
+            if (entry.next == null) {
+                break;
+            } else {
+                entry = entry.next;
+            }
+        }
```

Comment:
Do we need this while loop if it is always true?

Reply:
what do you mean? i am iterating over all nodes while statement true and if some condition i stop loop?

Reply:
Then you can change true to entry.next! = null. And change cycle to do-while